File: front-end/src/types/address.ts
// Define interfaces for request/response types
export interface IAddressRequest {
  receiverName: string;
  receiverPhone: string;
  province: string;
  district: string;
  ward: string;
  detail: string;
  isDefault?: boolean;
}

export interface IAddressResponse {
  id: string;
  receiverName: string;
  receiverPhone: string;
  province: string;
  district: string;
  ward: string;
  detail: string;
  isDefault: boolean;
}

File: back-end/src/main/java/javaweb/my_project/entities/Address.java
package javaweb.my_project.entities;

import com.fasterxml.jackson.annotation.JsonBackReference;
import com.fasterxml.jackson.annotation.JsonIgnore;
import jakarta.persistence.*;
import lombok.*;
import lombok.experimental.FieldDefaults;

@Entity
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
@FieldDefaults(level = AccessLevel.PRIVATE)
public class Address {
    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    String id;
    @Column(nullable = false)
    String province;
    @Column(nullable = false)
    String ward;
    @Column(nullable = false)
    String detail;
    @Column(nullable = false)
    @JsonIgnore
    Boolean isDefault;
    @Column(nullable = false)
    String receiverName;
    @Column(nullable = false)
    String receiverPhone;

    @OneToOne(mappedBy = "address")
    @JsonIgnore
    Farmer farmer;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "account_id")
    @JsonBackReference
    Account account;

}
